serving/purgatorio/select-and-train-machine-learning-models/tools-and-libraries/tools-and-libraries.md

```diff
@@ -1,5 +1,321 @@
+    + [Data Collection](#Data-Collection)
+    + [Data Preparation](#Data-Preparation)
+    + [Data Visualization](#Data-Visualization)
+    + [Machine Learning](#Machine-Learning)
+    + [Deep Learning](#Deep-Learning)
+    + [Hyperparameter Tuning](#Hyperparameter-Tuning)
+    + [Serving Models](#Serving-Models)
+    + [Miscellaneous](#Miscellaneous)
+
+- [Production Ready Tools](#Production-Ready-Tools)
+    + [Computer Vision](#Computer-vision)
+    + [Natural Language](#Natural-Language)
+    + [Time Series Forecasting](#Time-Series-Forecasting)
+
+- [Conclusions](#Conclusions)
+
+# How this guide is designed
+
+This guide is designed in the same fashion as the **Purgatorio** structure, so you find a subsection for each phase of the Data Science process.
+
+For _each section_, we list for you different libraries and the best resources to learn them.
+
+We'll give you some general tips to learn effectively and develop rock-solid foundations, that you can rely on to address and solve Data Science problems in the complexity of the real world (which is messy by definition).
+
+In this guide, you'll find a ton of different small tools, big libraries, and even entire frameworks, but be aware!
+
+As stated in the [Virgilio's Teaching Strategy](https://github.com/virgili0/Virgilio/blob/master/serving/paradiso/virgilio-teaching-strategy/virgilio-teaching-strategy.md) Guide, the best way to deal with the learning process is to leverage the "[Divide et Impera](https://en.wikipedia.org/wiki/Divide_and_rule)" principle, which states that every time you're tackling a problem (in this case, the _learning problem_) you should split it into smaller parts, easier to handle and understand. 
+
+# Solving the Learning Problem 
+
+The process of learning a tool and use it effectively **is equal to solving a problem**.
+
+We consider "solved" the problem when you can use the tool comfortably.
+
+So, you shouldn't try to learn all the features of a library in the same run, or you're going to be overwhelmed! 
+
+Every rich Python library contains thousands of different functions and utilities, but _you'll need a small subset to get the things done!_
+
+**So, Virgilio suggests the following approach**:
+
+- Choose a library from the ones that are widely used and feature-rich.
+- Get a general introduction through to Virgilio's resources listed in this guide.
+- Find a problem (similar but not equal to the introductory ones).
+- Dig into the problem, and get your hands dirty with the chosen library.
+- Learn _on the way to the problem solution_.
+- Solve the problem, trying to do it yourself.
+
+If you follow this approach, you take home the following 3 things:
+- **real problem experience**
+- **you feel comfortable coding with a subset of the functions of the tool**
+
+Believe it or not, it works like magic!
+
+Know you know how to solve a new _learning problem_.
+
+Now, how to choose from the myriad of tools and libraries available?
```

Comment:
From the above I would suggest this: and I have learnt from this method as well.

When trying to know about a library or framework or tool, use these two methods:
- search for them on Medium.com
- search for them on YouTube or another video/learning channel

Both these will show how others are using it or how else it can be used, much better than learning from the documentations initially. Eventually we can refer to all kinds of deep dive documentations etc...

Reply:
true i add this